tiledb/cf/netcdf_engine/_dim_converters.py

```diff
@@ -372,9 +453,21 @@ def get_values(
             The coordinates needed for querying the create TileDB dimension in the form
                 of a numpy array if sparse is ``True`` and a slice otherwise.
         """
+        if indexer.step not in {1, None}:
+            raise ValueError("Dimension indexer must have step size of 1.")
+        if indexer.start not in {None, 0}:
+            raise IndexError(
+                f"Cannot copy scalar data to TileDB dimension '{self.name}'. The NetCDF"
+                f" chunk starting at {indexer.start} is out of bounds."
+            )
+        if indexer.stop not in {None, 1}:
+            raise IndexError(
+                f"Cannot copy scalar data to TileDB dimension '{self.name}'. The NetCDF"
+                f" chunk ending at {indexer.stop} is out of bounds."
+            )
         if sparse:
             return np.array([0])
-        return slice(1)
+        return slice(0, 1)
```

Comment:
Or `return np.arange(1) if sparse else slice(1)` for consistency with `NetCDF4DimToDimConverter`